### ISSUE TITLE
decoupleR::get_collectri error

### ISSUE THREAD

**Issue by cf98** [saezlab/CollecTRI#21]
Hi Saezlab,

Thanks for developing this tool for TF analysis!
I am following [this vignette](https://www.bioconductor.org/packages/release/bioc/vignettes/decoupleR/inst/doc/tf_sc.html), and run into issues with command:
```
net <- decoupleR::get_collectri(organism='human', split_complexes=FALSE) 
```
The error message also came after a warning, here's the entire message: 
```
[2024-09-29 12:34:35] [WARN]    [OmnipathR] Accessing `collectri` as a static table: this is not the recommended way to access OmniPath data; it is only a backup plan for situations when our server or your computer is experiencing issues.
Error in if (.keep) . else select(., -!!evs_col) : 
  argument is of length zero
```

My R version is 4.4.1 (arm64, mac)
Current package versions:
```
           decoupleR                dplyr              ggplot2            OmnipathR            patchwork 
   "decoupleR 2.9.7"        "dplyr 1.1.4"      "ggplot2 3.5.1"  "OmnipathR 3.13.26"    "patchwork 1.3.0" 
            pheatmap              remotes               Seurat         SeuratObject                   sp 
   "pheatmap 1.0.12"      "remotes 2.5.0"       "Seurat 5.1.0" "SeuratObject 5.0.2"           "sp 2.1.4" 
              tibble                tidyr 
      "tibble 3.2.1"        "tidyr 1.3.1" 
```

Can you help on this?
Thanks!

**Comment by roxUCL** [comment]
Hello,
same here. Same error same R 4.4.1 (arm64, mac) and same installed package versions [1] pheatmap_1.0.12   dplyr_1.1.4       devtools_2.4.5    usethis_3.0.0     decoupleR_2.9.7   OmnipathR_3.13.26.
Could you kindly help on this?

**Comment by gabora** [comment]
Hi @deeenes , 
this seems to me an Omnipath issue, or what do you think?

**Comment by deeenes** [comment]
See https://github.com/saezlab/OmnipathR/issues/106

TLDR:

```
library(remotes)
remotes::install_github('saezlab/OmnipathR')

library(OmnipathR)
library(decoupleR)

omnipath_cache_wipe()

net = get_collectri(organism='human', split_complexes=FALSE)

# if it still fails, please share the OmnipathR log:

omnipath_log()
```

**Comment by roxUCL** [comment]
Thanks for your supoport in this. 

I think the error is still there 
this is omnipath_log():
[2024-09-30 12:18:53] [INFO]    [OmnipathR] Welcome to OmnipathR!
[2024-09-30 12:18:53] [TRACE]   [OmnipathR] Reading JSON from `/Users/rossellarispoli/Library/Caches/OmnipathR/cache.json` (encoding: UTF-8).
[2024-09-30 12:18:53] [TRACE]   [OmnipathR] JSON validation successful: TRUE
[2024-09-30 12:18:53] [INFO]    [OmnipathR] Initialized cache: `/Users/rossellarispoli/Library/Caches/OmnipathR`.
[2024-09-30 12:18:53] [TRACE]   [OmnipathR] Reading JSON from `/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/library/OmnipathR/db/db_def.json` (encoding: UTF-8).
[2024-09-30 12:18:53] [TRACE]   [OmnipathR] JSON validation successful: TRUE
[2024-09-30 12:18:53] [TRACE]   [OmnipathR] Reading JSON from `/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/library/OmnipathR/internal/magic_bytes.json` (encoding: UTF-8).
[2024-09-30 12:18:53] [TRACE]   [OmnipathR] JSON validation successful: TRUE
[2024-09-30 12:18:54] [TRACE]   [OmnipathR] Reading JSON from `/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/library/OmnipathR/internal/urls.json` (encoding: UTF-8).
[2024-09-30 12:18:54] [TRACE]   [OmnipathR] JSON validation successful: TRUE
[2024-09-30 12:18:54] [TRACE]   [OmnipathR] Reading JSON from `/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/library/OmnipathR/internal/id_types.json` (encoding: UTF-8).
[2024-09-30 12:18:54] [TRACE]   [OmnipathR] JSON validation successful: TRUE
[2024-09-30 12:19:13] [INFO]    [OmnipathR] Loading database `Ensembl and OMA organism names`.
[2024-09-30 12:19:13] [INFO]    [OmnipathR] Looking up in cache `https://www.ensembl.org/info/about/species.html`: key=7332486db7400730697234bad76ca0c8e4d00799, latest version=1.
[2024-09-30 12:19:13] [TRACE]   [OmnipathR] Cache file path: /Users/rossellarispoli/Library/Caches/OmnipathR/7332486db7400730697234bad76ca0c8e4d00799-1.html
[2024-09-30 12:19:13] [TRACE]   [OmnipathR] Looking up in cache: `https://omabrowser.org/All/oma-species.txt`.
[2024-09-30 12:19:13] [TRACE]   [OmnipathR] Loaded data from RDS `/Users/rossellarispoli/Library/Caches/OmnipathR/30e690cbb55dfc63b5903ab337f34ffc2f4be397-1.rds`.
[2024-09-30 12:19:14] [INFO]    [OmnipathR] Loading database `Ensembl and OMA organism names`.
[2024-09-30 12:19:14] [INFO]    [OmnipathR] Looking up in cache `https://www.ensembl.org/info/about/species.html`: key=7332486db7400730697234bad76ca0c8e4d00799, latest version=1.
[2024-09-30 12:19:14] [TRACE]   [OmnipathR] Cache file path: /Users/rossellarispoli/Library/Caches/OmnipathR/7332486db7400730697234bad76ca0c8e4d00799-1.html
[2024-09-30 12:19:14] [TRACE]   [OmnipathR] Looking up in cache: `https://omabrowser.org/All/oma-species.txt`.
[2024-09-30 12:19:14] [TRACE]   [OmnipathR] Loaded data from RDS `/Users/rossellarispoli/Library/Caches/OmnipathR/30e690cbb55dfc63b5903ab337f34ffc2f4be397-1.rds`.
[2024-09-30 12:19:14] [WARN]    [OmnipathR] Accessing `collectri` as a static table: this is not the recommended way to access OmniPath data; it is only a backup plan for situations when our server or your computer is experiencing issues.
[2024-09-30 12:19:14] [TRACE]   [OmnipathR] Looking up in cache: `http://no-tls.static.omnipathdb.org/resources/interactions_collectri_9606.tsv.gz`
[2024-09-30 12:19:15] [TRACE]   [OmnipathR] Loaded data from RDS `/Users/rossellarispoli/Library/Caches/OmnipathR/a2bbed8b281906d09d7cf983d102402e5de5da8c-1.rds`.
[2024-09-30 12:19:15] [INFO]    [OmnipathR] Loaded from cache: `http://no-tls.static.omnipathdb.org/resources/interactions_collectri_9606.tsv.gz`
[2024-09-30 12:19:15] [TRACE]   [OmnipathR] Converting JSON column `evidences` to list.
[2024-09-30 12:19:17] [TRACE]   [OmnipathR] Restricting interaction records to datasets: collectri; and resources: any
[2024-09-30 12:25:29] [INFO]    [OmnipathR] Loading database `Ensembl and OMA organism names`.
[2024-09-30 12:25:29] [INFO]    [OmnipathR] Looking up in cache `https://www.ensembl.org/info/about/species.html`: key=7332486db7400730697234bad76ca0c8e4d00799, latest version=1.
[2024-09-30 12:25:29] [TRACE]   [OmnipathR] Cache file path: /Users/rossellarispoli/Library/Caches/OmnipathR/7332486db7400730697234bad76ca0c8e4d00799-1.html
[2024-09-30 12:25:29] [TRACE]   [OmnipathR] Looking up in cache: `https://omabrowser.org/All/oma-species.txt`.
[2024-09-30 12:25:29] [TRACE]   [OmnipathR] Loaded data from RDS `/Users/rossellarispoli/Library/Caches/OmnipathR/30e690cbb55dfc63b5903ab337f34ffc2f4be397-1.rds`.
[2024-09-30 12:28:27] [INFO]    [OmnipathR] Loading database `Ensembl and OMA organism names`.
[2024-09-30 12:28:27] [INFO]    [OmnipathR] Looking up in cache `https://www.ensembl.org/info/about/species.html`: key=7332486db7400730697234bad76ca0c8e4d00799, latest version=1.
[2024-09-30 12:28:27] [TRACE]   [OmnipathR] Cache file path: /Users/rossellarispoli/Library/Caches/OmnipathR/7332486db7400730697234bad76ca0c8e4d00799-1.html
[2024-09-30 12:28:27] [TRACE]   [OmnipathR] Looking up in cache: `https://omabrowser.org/All/oma-species.txt`.
[2024-09-30 12:28:27] [TRACE]   [OmnipathR] Loaded data from RDS `/Users/rossellarispoli/Library/Caches/OmnipathR/30e690cbb55dfc63b5903ab337f34ffc2f4be397-1.rds`.
[2024-09-30 12:28:28] [INFO]    [OmnipathR] Loading database `Ensembl and OMA organism names`.
[2024-09-30 12:28:28] [INFO]    [OmnipathR] Looking up in cache `https://www.ensembl.org/info/about/species.html`: key=7332486db7400730697234bad76ca0c8e4d00799, latest version=1.
[2024-09-30 12:28:28] [TRACE]   [OmnipathR] Cache file path: /Users/rossellarispoli/Library/Caches/OmnipathR/7332486db7400730697234bad76ca0c8e4d00799-1.html
[2024-09-30 12:28:28] [TRACE]   [OmnipathR] Looking up in cache: `https://omabrowser.org/All/oma-species.txt`.
[2024-09-30 12:28:28] [TRACE]   [OmnipathR] Loaded data from RDS `/Users/rossellarispoli/Library/Caches/OmnipathR/30e690cbb55dfc63b5903ab337f34ffc2f4be397-1.rds`.
[2024-09-30 12:28:29] [WARN]    [OmnipathR] Accessing `collectri` as a static table: this is not the recommended way to access OmniPath data; it is only a backup plan for situations when our server or your computer is experiencing issues.
[2024-09-30 12:28:29] [TRACE]   [OmnipathR] Looking up in cache: `http://no-tls.static.omnipathdb.org/resources/interactions_collectri_9606.tsv.gz`
[2024-09-30 12:28:29] [TRACE]   [OmnipathR] Loaded data from RDS `/Users/rossellarispoli/Library/Caches/OmnipathR/a2bbed8b281906d09d7cf983d102402e5de5da8c-1.rds`.
[2024-09-30 12:28:29] [INFO]    [OmnipathR] Loaded from cache: `http://no-tls.static.omnipathdb.org/resources/interactions_collectri_9606.tsv.gz`
[2024-09-30 12:28:29] [TRACE]   [OmnipathR] Converting JSON column `evidences` to list.
[2024-09-30 12:28:31] [TRACE]   [OmnipathR] Restricting interaction records to datasets: collectri; and resources: any
[2024-09-30 12:41:57] [INFO]    [OmnipathR] Loading database `Ensembl and OMA organism names`.
[2024-09-30 12:41:57] [INFO]    [OmnipathR] Looking up in cache `https://www.ensembl.org/info/about/species.html`: key=7332486db7400730697234bad76ca0c8e4d00799, latest version=1.
[2024-09-30 12:41:57] [TRACE]   [OmnipathR] Cache file path: /Users/rossellarispoli/Library/Caches/OmnipathR/7332486db7400730697234bad76ca0c8e4d00799-1.html
[2024-09-30 12:41:57] [TRACE]   [OmnipathR] Looking up in cache: `https://omabrowser.org/All/oma-species.txt`.
[2024-09-30 12:41:57] [TRACE]   [OmnipathR] Loaded data from RDS `/Users/rossellarispoli/Library/Caches/OmnipathR/30e690cbb55dfc63b5903ab337f34ffc2f4be397-1.rds`.
[2024-09-30 12:41:58] [INFO]    [OmnipathR] Loading database `Ensembl and OMA organism names`.
[2024-09-30 12:41:58] [INFO]    [OmnipathR] Looking up in cache `https://www.ensembl.org/info/about/species.html`: key=7332486db7400730697234bad76ca0c8e4d00799, latest version=1.
[2024-09-30 12:41:58] [TRACE]   [OmnipathR] Cache file path: /Users/rossellarispoli/Library/Caches/OmnipathR/7332486db7400730697234bad76ca0c8e4d00799-1.html
[2024-09-30 12:41:58] [TRACE]   [OmnipathR] Looking up in cache: `https://omabrowser.org/All/oma-species.txt`.
[2024-09-30 12:41:58] [TRACE]   [OmnipathR] Loaded data from RDS `/Users/rossellarispoli/Library/Caches/OmnipathR/30e690cbb55dfc63b5903ab337f34ffc2f4be397-1.rds`.
[2024-09-30 12:41:58] [WARN]    [OmnipathR] Accessing `collectri` as a static table: this is not the recommended way to access OmniPath data; it is only a backup plan for situations when our server or your computer is experiencing issues.
[2024-09-30 12:41:58] [TRACE]   [OmnipathR] Looking up in cache: `http://no-tls.static.omnipathdb.org/resources/interactions_collectri_9606.tsv.gz`
[2024-09-30 12:41:59] [TRACE]   [OmnipathR] Loaded data from RDS `/Users/rossellarispoli/Library/Caches/OmnipathR/a2bbed8b281906d09d7cf983d102402e5de5da8c-1.rds`.
[2024-09-30 12:41:59] [INFO]    [OmnipathR] Loaded from cache: `http://no-tls.static.omnipathdb.org/resources/interactions_collectri_9606.tsv.gz`
[2024-09-30 12:41:59] [TRACE]   [OmnipathR] Converting JSON column `evidences` to list.
[2024-09-30 12:42:03] [TRACE]   [OmnipathR] Restricting interaction records to datasets: collectri; and resources: any
[2024-09-30 13:28:49] [TRACE]   [OmnipathR] Reading JSON from `/Users/rossellarispoli/Library/Caches/OmnipathR/cache.json` (encoding: UTF-8).
[2024-09-30 13:28:49] [TRACE]   [OmnipathR] JSON validation successful: TRUE
[2024-09-30 13:28:49] [SUCCESS] [OmnipathR] Removing all cache contents from `/Users/rossellarispoli/Library/Caches/OmnipathR`.
[2024-09-30 13:28:53] [INFO]    [OmnipathR] Loading database `Ensembl and OMA organism names`.
[2024-09-30 13:28:53] [TRACE]   [OmnipathR] Reading JSON from `/Users/rossellarispoli/Library/Caches/OmnipathR/cache.json` (encoding: UTF-8).
[2024-09-30 13:28:53] [TRACE]   [OmnipathR] JSON validation successful: TRUE
[2024-09-30 13:28:53] [INFO]    [OmnipathR] Looking up in cache `https://www.ensembl.org/info/about/species.html`: key=7332486db7400730697234bad76ca0c8e4d00799, no version available.
[2024-09-30 13:28:53] [TRACE]   [OmnipathR] Reading JSON from `/Users/rossellarispoli/Library/Caches/OmnipathR/cache.json` (encoding: UTF-8).
[2024-09-30 13:28:53] [TRACE]   [OmnipathR] JSON validation successful: TRUE
[2024-09-30 13:28:53] [INFO]    [OmnipathR] Created new version for cache record 7332486db7400730697234bad76ca0c8e4d00799: version 1.
[2024-09-30 13:28:53] [TRACE]   [OmnipathR] Cache file path: /Users/rossellarispoli/Library/Caches/OmnipathR/7332486db7400730697234bad76ca0c8e4d00799-1.html
[2024-09-30 13:28:53] [INFO]    [OmnipathR] Retrieving URL: `https://www.ensembl.org/info/about/species.html`
[2024-09-30 13:28:53] [TRACE]   [OmnipathR] Attempt 1/3: `https://www.ensembl.org/info/about/species.html`
[2024-09-30 13:28:54] [TRACE]   [OmnipathR] HTTP 200
[2024-09-30 13:28:54] [INFO]    [OmnipathR] Download ready [key=7332486db7400730697234bad76ca0c8e4d00799, version=1]
[2024-09-30 13:28:54] [TRACE]   [OmnipathR] Reading JSON from `/Users/rossellarispoli/Library/Caches/OmnipathR/cache.json` (encoding: UTF-8).
[2024-09-30 13:28:54] [TRACE]   [OmnipathR] JSON validation successful: TRUE
[2024-09-30 13:28:54] [INFO]    [OmnipathR] Cache item `7332486db7400730697234bad76ca0c8e4d00799` version 1: status changed from `unknown` to `ready`.
[2024-09-30 13:28:55] [TRACE]   [OmnipathR] Looking up in cache: `https://omabrowser.org/All/oma-species.txt`.
[2024-09-30 13:28:55] [INFO]    [OmnipathR] Cache record does not exist: `https://omabrowser.org/All/oma-species.txt`
[2024-09-30 13:28:55] [TRACE]   [OmnipathR] Could not find in cache, initiating download: `https://omabrowser.org/All/oma-species.txt`.
[2024-09-30 13:28:55] [TRACE]   [OmnipathR] Reading JSON from `/Users/rossellarispoli/Library/Caches/OmnipathR/cache.json` (encoding: UTF-8).
[2024-09-30 13:28:55] [TRACE]   [OmnipathR] JSON validation successful: TRUE
[2024-09-30 13:28:55] [TRACE]   [OmnipathR] Reading JSON from `/Users/rossellarispoli/Library/Caches/OmnipathR/cache.json` (encoding: UTF-8).
[2024-09-30 13:28:55] [TRACE]   [OmnipathR] JSON validation successful: TRUE
[2024-09-30 13:28:55] [INFO]    [OmnipathR] Cache item `30e690cbb55dfc63b5903ab337f34ffc2f4be397` version 1: status changed from `unknown` to `started`.
[2024-09-30 13:28:55] [TRACE]   [OmnipathR] Exporting object to RDS: `/Users/rossellarispoli/Library/Caches/OmnipathR/30e690cbb55dfc63b5903ab337f34ffc2f4be397-1.rds`.
[2024-09-30 13:28:55] [INFO]    [OmnipathR] Retrieving URL: `https://omabrowser.org/All/oma-species.txt`
[2024-09-30 13:28:55] [TRACE]   [OmnipathR] Attempt 1/3: `https://omabrowser.org/All/oma-species.txt`
[2024-09-30 13:28:57] [TRACE]   [OmnipathR] Exported RDS to `/Users/rossellarispoli/Library/Caches/OmnipathR/30e690cbb55dfc63b5903ab337f34ffc2f4be397-1.rds`.
[2024-09-30 13:28:57] [INFO]    [OmnipathR] Download ready [key=30e690cbb55dfc63b5903ab337f34ffc2f4be397, version=1]
[2024-09-30 13:28:57] [TRACE]   [OmnipathR] Reading JSON from `/Users/rossellarispoli/Library/Caches/OmnipathR/cache.json` (encoding: UTF-8).
[2024-09-30 13:28:57] [TRACE]   [OmnipathR] JSON validation successful: TRUE
[2024-09-30 13:28:57] [INFO]    [OmnipathR] Cache item `30e690cbb55dfc63b5903ab337f34ffc2f4be397` version 1: status changed from `started` to `ready`.
[2024-09-30 13:28:58] [INFO]    [OmnipathR] Loading database `Ensembl and OMA organism names`.
[2024-09-30 13:28:58] [INFO]    [OmnipathR] Looking up in cache `https://www.ensembl.org/info/about/species.html`: key=7332486db7400730697234bad76ca0c8e4d00799, latest version=1.
[2024-09-30 13:28:58] [TRACE]   [OmnipathR] Cache file path: /Users/rossellarispoli/Library/Caches/OmnipathR/7332486db7400730697234bad76ca0c8e4d00799-1.html
[2024-09-30 13:28:58] [TRACE]   [OmnipathR] Looking up in cache: `https://omabrowser.org/All/oma-species.txt`.
[2024-09-30 13:28:58] [TRACE]   [OmnipathR] Loaded data from RDS `/Users/rossellarispoli/Library/Caches/OmnipathR/30e690cbb55dfc63b5903ab337f34ffc2f4be397-1.rds`.
[2024-09-30 13:29:00] [WARN]    [OmnipathR] Accessing `collectri` as a static table: this is not the recommended way to access OmniPath data; it is only a backup plan for situations when our server or your computer is experiencing issues.
[2024-09-30 13:29:00] [TRACE]   [OmnipathR] Looking up in cache: `http://no-tls.static.omnipathdb.org/resources/interactions_collectri_9606.tsv.gz`
[2024-09-30 13:29:00] [INFO]    [OmnipathR] Cache record does not exist: `http://no-tls.static.omnipathdb.org/resources/interactions_collectri_9606.tsv.gz`
[2024-09-30 13:29:00] [TRACE]   [OmnipathR] Attempting `http://no-tls.static.omnipathdb.org/resources/interactions_collectri_9606.tsv.gz`
[2024-09-30 13:29:00] [INFO]    [OmnipathR] Retrieving URL: `http://no-tls.static.omnipathdb.org/resources/interactions_collectri_9606.tsv.gz`
[2024-09-30 13:29:00] [TRACE]   [OmnipathR] Attempt 1/3: `http://no-tls.static.omnipathdb.org/resources/interactions_collectri_9606.tsv.gz`
[2024-09-30 13:29:01] [INFO]    [OmnipathR] Successfully retrieved: `http://no-tls.static.omnipathdb.org/resources/interactions_collectri_9606.tsv.gz`
[2024-09-30 13:29:01] [TRACE]   [OmnipathR] Reading JSON from `/Users/rossellarispoli/Library/Caches/OmnipathR/cache.json` (encoding: UTF-8).
[2024-09-30 13:29:01] [TRACE]   [OmnipathR] JSON validation successful: TRUE
[2024-09-30 13:29:01] [TRACE]   [OmnipathR] Reading JSON from `/Users/rossellarispoli/Library/Caches/OmnipathR/cache.json` (encoding: UTF-8).
[2024-09-30 13:29:01] [TRACE]   [OmnipathR] JSON validation successful: TRUE
[2024-09-30 13:29:01] [INFO]    [OmnipathR] Cache item `a2bbed8b281906d09d7cf983d102402e5de5da8c` version 1: status changed from `unknown` to `started`.
[2024-09-30 13:29:01] [TRACE]   [OmnipathR] Exporting object to RDS: `/Users/rossellarispoli/Library/Caches/OmnipathR/a2bbed8b281906d09d7cf983d102402e5de5da8c-1.rds`.
[2024-09-30 13:29:01] [TRACE]   [OmnipathR] Exported RDS to `/Users/rossellarispoli/Library/Caches/OmnipathR/a2bbed8b281906d09d7cf983d102402e5de5da8c-1.rds`.
[2024-09-30 13:29:01] [INFO]    [OmnipathR] Download ready [key=a2bbed8b281906d09d7cf983d102402e5de5da8c, version=1]
[2024-09-30 13:29:01] [TRACE]   [OmnipathR] Reading JSON from `/Users/rossellarispoli/Library/Caches/OmnipathR/cache.json` (encoding: UTF-8).
[2024-09-30 13:29:01] [TRACE]   [OmnipathR] JSON validation successful: TRUE
[2024-09-30 13:29:01] [INFO]    [OmnipathR] Cache item `a2bbed8b281906d09d7cf983d102402e5de5da8c` version 1: status changed from `started` to `ready`.
[2024-09-30 13:29:01] [TRACE]   [OmnipathR] Converting JSON column `evidences` to list.
[2024-09-30 13:29:04] [TRACE]   [OmnipathR] Restricting interaction records to datasets: collectri; and resources: any
[2024-09-30 13:29:25] [INFO]    [OmnipathR] Loading database `Ensembl and OMA organism names`.
[2024-09-30 13:29:25] [INFO]    [OmnipathR] Looking up in cache `https://www.ensembl.org/info/about/species.html`: key=7332486db7400730697234bad76ca0c8e4d00799, latest version=1.
[2024-09-30 13:29:25] [TRACE]   [OmnipathR] Cache file path: /Users/rossellarispoli/Library/Caches/OmnipathR/7332486db7400730697234bad76ca0c8e4d00799-1.html
[2024-09-30 13:29:26] [TRACE]   [OmnipathR] Looking up in cache: `https://omabrowser.org/All/oma-species.txt`.
[2024-09-30 13:29:26] [TRACE]   [OmnipathR] Loaded data from RDS `/Users/rossellarispoli/Library/Caches/OmnipathR/30e690cbb55dfc63b5903ab337f34ffc2f4be397-1.rds`.
[2024-09-30 13:29:26] [INFO]    [OmnipathR] Loading database `Ensembl and OMA organism names`.
[2024-09-30 13:29:26] [INFO]    [OmnipathR] Looking up in cache `https://www.ensembl.org/info/about/species.html`: key=7332486db7400730697234bad76ca0c8e4d00799, latest version=1.
[2024-09-30 13:29:26] [TRACE]   [OmnipathR] Cache file path: /Users/rossellarispoli/Library/Caches/OmnipathR/7332486db7400730697234bad76ca0c8e4d00799-1.html
[2024-09-30 13:29:26] [TRACE]   [OmnipathR] Looking up in cache: `https://omabrowser.org/All/oma-species.txt`.
[2024-09-30 13:29:26] [TRACE]   [OmnipathR] Loaded data from RDS `/Users/rossellarispoli/Library/Caches/OmnipathR/30e690cbb55dfc63b5903ab337f34ffc2f4be397-1.rds`.
[2024-09-30 13:29:27] [WARN]    [OmnipathR] Accessing `collectri` as a static table: this is not the recommended way to access OmniPath data; it is only a backup plan for situations when our server or your computer is experiencing issues.
[2024-09-30 13:29:27] [TRACE]   [OmnipathR] Looking up in cache: `http://no-tls.static.omnipathdb.org/resources/interactions_collectri_9606.tsv.gz`
[2024-09-30 13:29:27] [TRACE]   [OmnipathR] Loaded data from RDS `/Users/rossellarispoli/Library/Caches/OmnipathR/a2bbed8b281906d09d7cf983d102402e5de5da8c-1.rds`.
[2024-09-30 13:29:27] [INFO]    [OmnipathR] Loaded from cache: `http://no-tls.static.omnipathdb.org/resources/interactions_collectri_9606.tsv.gz`
[2024-09-30 13:29:27] [TRACE]   [OmnipathR] Converting JSON column `evidences` to list.
[2024-09-30 13:29:30] [TRACE]   [OmnipathR] Restricting interaction records to datasets: collectri; and resources: any

**Comment by deeenes** [comment]
Hi @roxUCL,

Have you done both of these?

- Updating `OmnipathR` from GitHub: `install_github('saezlab/OmnipathR')`
- Wiping its cache: `omnipath_cache_wipe()`

If so, can you please share the version of the loaded package, and the output of the `oma_organisms` function?

```
library(OmnipathR)

packageVersion('OmnipathR')

oma_organisms()
```

**Comment by roxUCL** [comment]
Hello, 
thanks for your support yes I have installed from the github page and this is the version ‘3.13.26’. 

For the oma_ornaism() it return a tibble of2,927 × 5 do you want me to save it into a file and share with you or just check the human organism?

**Comment by deeenes** [comment]
No need for the whole tibble, just copy-paste the header and first row, like this:

```
 oma_organisms()
# A tibble: 2,927 × 6
   oma_code ncbi_tax_id latin_name                                                                                          genome_source  oma_version                                                    gtdb_tax_id
   <chr>          <dbl> <chr>                                                                                               <chr>          <chr>                                                                <dbl>
 1 ABSGL           4829 Absidia glauca                                                                                      EnsemblGenomes Ensembl Fungi 48; AG_v1                                                 NA

```

The column layout of this dataset has been changed by its provider, this caused the error for 3-4 other users, hence we first have to make sure it's correctly processed in your system. If it is, then you have a different error, and we'll look for it further.

**Comment by roxUCL** [comment]
oma_organisms()
# A tibble: 2,927 × 5
   oma_code  oma_tax_id ncbi_tax_id                                                                                         gtdb_genome    latin_name                                                
   <chr>          <dbl> <chr>                                                                                               <chr>          <chr>                                                     
 1 ABSGL           4829 Absidia glauca                                                                                      EnsemblGenomes Ensembl Fungi 48; AG_v1                                   
 2 ACACA        1257118 Acanthamoeba castellanii str. Neff                                                                  Genbank        Genbank; Acastellanii.strNEFF v1; GCF_000313135.1; 07-OCT…
 3 ACAM1     -644103924 Acaryochloris marina (strain MBIC 11017)

**Comment by deeenes** [comment]
@roxUCL This is the new content, but old header: you need to get rid of the old cache file, easiest way is to wipe the cache completely, still before calling `oma_organisms`. Then check if you get the same column layout as in my comment above.

```
library(OmnipathR)

omnipath_cache_wipe()

oma_organisms()
```

**Comment by deeenes** [comment]
I can explain the 5 columns layout we see in your environment only if I assume the new file has been saved in the cache by the old code, and then read from there also by the old code. This definitely happened before you updated OmnipathR. But the absence of `gtdb_tax_id` column suggests that still the old code is active in your current session. However, `packageVersion('OmnipathR') == '3.13.26'` should guarantee that the code is up-to-date. I assume, since the reinstallation and cache wipe you've started a new R session?

First, let's check the loaded code is indeed up-to-date:

```
print(oma_organisms)
```

Then let's see all the details about the cache item:

```
omnipathr.env$cache[['30e690cbb55dfc63b5903ab337f34ffc2f4be397']]
```

Though the wipe above should've removed this file, so it should be a very fresh copy, maybe you can try manually removing and loading it again in a fresh session:

```
rm /Users/rossellarispoli/Library/Caches/OmnipathR/30e690cbb55dfc63b5903ab337f34ffc2f4be397*
```

```
library(OmnipathR)
omnipath_set_console_loglevel('trace')
omnipath_cache_clean_db()
oma_organisms()
```

**Comment by cf98** [comment]
> I think I have tried this before with no success.
> 
> I have done it now again and again it does not seams to me that there is a change
> 
> ```
> > library(OmnipathR)
> > 
> > omnipath_cache_wipe()
> [2024-09-30 14:47:21] [SUCCESS] [OmnipathR] Removing all cache contents from `/Users/rossellarispoli/Library/Caches/OmnipathR`.
> > 
>  oma_organisms()
> # A tibble: 2,927 × 5
>    oma_code  oma_tax_id ncbi_tax_id                                                                                         gtdb_genome    latin_name                                                
>    <chr>          <dbl> <chr>                                                                                               <chr>          <chr>                                                     
>  1 ABSGL           4829 Absidia glauca                                                                                      EnsemblGenomes Ensembl Fungi 48; AG_v1                                   
>  2 ACACA        1257118 Acanthamoeba castellanii str. Neff                                                                  Genbank        Genbank; Acastellanii.strNEFF v1; GCF_000313135.1; 07-OCT…
>  3 ACAM1     -644103924 Acaryochloris marina (strain MBIC 11017) 
> ```
> 
> should I amnually change the header?

Same here! I have also tried wipe out old cache files again and again, it didn't work unfortunately. Thanks for sharing the issues you run into!

**Comment by deeenes** [comment]
@cf98 I can tell you the same as 2 comments above: the question is why and how the old code operates here when the package is supposed to be updated. We should check what is the loaded code:

```
library(OmnipathR)

print(oma_organisms)
```

And a `sessionInfo` would be helpful too:

```
library(OmnipathR)

sessionInfo()
```

**Comment by roxUCL** [comment]
thanks @deeenes removing the file OmnipathR cache file worked for me

**Comment by deeenes** [comment]
@roxUCL I'm glad to hear! It means `omnipath_cache_wipe` didn't work, that's really weird.

@cf98 - Then maybe it will work for you too -- see 4 comments above, how an individual cache file can be removed.
